### PR TITLE
Update overlay components to new theme system

### LINKED ISSUE #131: Update Dialog docs page to include fully working example
Per #123:

> update the documentation to include a fully working example (button to open, able to close, etc). I'd normally say to make that change now, however, creating classes inside the docz playground is broken. So let's keep an eye on this issue [pedronauck/docz#567](https://github.com/pedronauck/docz/issues/567) and once it's resolved setup our documentation to be a bit more robust.

### LINKED ISSUE #171: Cleanup: Dialog

### LINKED ISSUE #172: Cleanup: Tooltip

### LINKED ISSUE #205: Can't close dialog on docs site
no-close

:(

### PR DESCRIPTION
**Pull request by lederer**
## Target Branch
Note this is a request to merge to `feature/theme`, not `develop`. All theme-related PRs should target `feature/theme` until the theme upgrade is complete.

## Overview

Updates the following components to the new theme system:

- Tooltip
- Dialog

Updates the Dialog docs page with a fully working example and clearer instructions for how to close the dialog.

### Checklist

- [x] Relevant documentation pages have been created or updated
- [ ] Description of PR is in an appropriate section of the changelog and grouped with similar changes if possible
- [x] Relevant status has been updated on the status page

### Upgrade instructions

If there are any of the following in this PR, provide proper instructions on how to upgrade:

- [ ] Changes to the prop names or types of existing components
- [ ] Changes in intended behavior of existing component props
- [ ] Changes in the theme file's structure
- [ ] A required version bump to a non-dev dependency of the project

### Notes

Did a some drive-by tweaks to Callout, Breadcrumbs, Button.

This PR does not change the underlying tooltip and dialog libraries used by Tooltip and Dialog.

## Testing Instructions

- Fiddle with the corresponding docs pages

Closes #171 
Closes #172 
Closes #131 
Closes #205

### CODE REVIEW

File: packages/core/components/dialog/index.js [hunk 1]
```diff
@@ -1,24 +1,33 @@
 import React from "react";
 import PropTypes from "prop-types";
+import { transparentize } from "polished";
```
Comment (designmatty) [inline packages/core/components/dialog/index.js]: remove

File: packages/core/components/dialog/index.js [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
   width: 100%;
   height: 100%;
   overflow: auto;
   -webkit-overflow-scrolling: touch;
   outline: none;
+
+  minWidth: ${themeGet("dialog.minWidths.minWidth")};
+  maxWidth: ${themeGet("dialog.maxWidths.maxWidth")};
+  minHeight: ${themeGet("dialog.minHeights.minHeight")};
+  maxHeight: ${themeGet("dialog.maxHeights.maxHeight")};
```
Comment (designmatty) [inline packages/core/components/dialog/index.js]: incorrect syntax

File: packages/core/components/dialog/index.js [hunk 3]
```diff
@@ -70,7 +79,7 @@ const Dialog = ({
     display: 'flex',
     justifyContent: 'center',
     alignItems: 'center',
-    backgroundColor: `rgba(0, 0, 0, ${themeGet(`opacity.${opacity}`, opacity)(props)})`,
+    backgroundColor: `rgba(0, 0, 0, ${opacity})`,
```
Comment (designmatty) [inline packages/core/components/dialog/index.js:81]: we might consider allowing user-defined colors here.

**Comment by lederer** [comment]
Just added an unrelated fix for Checkbox and Radio vertical alignment.

File: packages/core/components/dialog/index.mdx [hunk 4]
```diff
@@ -7,47 +7,89 @@ route: /overlays/dialog
...
-certain styles can be configured via styled-system props (see table below) and
-the rest are hard-coded opinions within the `Dialog` component.
+certain styles can be configured via the theme properties (see below), or their
+corresponding styled-system props, and the rest are hard-coded opinions within
+the `Dialog` component.
 
-Notable props include `opacity` for adjusting the scrim opacity, `boxShadow` for
-setting the dialog drop shadow, and `closeTimeoutMS` for adjusting the transition
-timing. Transitions are hard-coded for now. Only the timing is adjustable.
+Overlay opacity can be adjusted via the `opacity` prop.
```
Comment (designmatty) [inline packages/core/components/dialog/index.mdx:39]: If you end up making the suggested change about user-defined colors, update this as well.

**Comment by lederer** [comment]
Just added an unrelated fix for Grouper alignment.

File: packages/core/components/button/index.js [hunk 5]
```diff
@@ -49,7 +45,7 @@ const ButtonLoading = styled.span`
   align-items: center;
   justify-content: center;
 
-  > ${ButtonIcon} {
+  > .icon {
```
Comment (designmatty) [inline packages/core/components/button/index.js:48]: Nothing wrong with this change, just curious about the decision behind it.

File: packages/core/components/dialog/index.mdx [hunk 6]
```diff
@@ -7,47 +7,89 @@ route: /overlays/dialog
 import { Playground, Props } from 'docz'
 import Dialog from './'
 import Button from '../../../core/components/button'
+import Heading from '../../../core/components/heading'
```
Comment (designmatty) [inline packages/core/components/dialog/index.mdx]: these imports can simply be `'../button'` and `'../heading'`

**Review by designmatty — CHANGES_REQUESTED** [review]

File: packages/core/components/tooltip/index.js [hunk 7]
```diff
@@ -1,38 +1,43 @@
 import React from "react";
 import PropTypes from "prop-types";
 import styled, { css } from "styled-components";
```
Comment (designmatty) [inline packages/core/components/tooltip/index.js]: css import not needed

File: packages/core/components/dialog/index.js (hunk 2 above)
Comment (lederer) [reply packages/core/components/dialog/index.js]: Gah! I keep doing that.

File: packages/core/components/dialog/index.js (hunk 3 above)
Comment (lederer) [reply packages/core/components/dialog/index.js:81]: I tried that – it's actually the source of that stray polished `transparentize` import – but couldn't get it working. Decided to punt for now. I'll open an issue.

File: packages/core/components/button/index.js (hunk 5 above)
Comment (lederer) [reply packages/core/components/button/index.js:48]: `ButtonIcon` was a `styled(Icon)`. I've opted to stop using that syntax entirely (in favor of `styled.div` or other built-in html elements), so I just used the `Icon` component directly in the jsx. Hence needed to give it a classname to refer to it here.

**Comment by lederer** [comment]
Thanks for catching all those. Ready for another look.

File: packages/core/components/dialog/index.js (hunk 3 above)
Comment (designmatty) [reply packages/core/components/dialog/index.js:81]: The following should work:

```
const overlayStyle = {
    ...
    background: transparentize(opacity, backdropColor)
}
```

And obviously, you would then need to pass `backdropColor` as a prop and pass a defaultProp value.

Comment (designmatty) [reply packages/core/components/dialog/index.js:81]: image
image

Comment (designmatty) [reply packages/core/components/dialog/index.js:81]: The biggest downside to this approach is the lack of access to the theme. That might require a larger rewrite.

Comment (lederer) [reply packages/core/components/dialog/index.js:81]: Yeah the problem I had was supporting the overlay color in the theme file too. Would have to add some logic to use the prop if present, else fallback on the theme. We do this a lot inside our styled component definitions, but not in a standalone `const`. I'm sure it's doable, I just hit a wall and so punted.

Comment (designmatty) [reply packages/core/components/dialog/index.js:81]: Okay, yeah that's fair. Agreed on the issue then.

**Review by designmatty — APPROVED** [review]
Looks good

**Pull request merged**